waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -225,14 +240,13 @@ func (wakuM *WakuMetadata) Connected(n network.Network, cc network.Conn) {
 		shard, err := wakuM.Request(wakuM.ctx, peerID)
 		if err == nil {
 			if shard == nil {
-				err = errors.New("no shard reported")
+				shouldDisconnect = false // No shard reported
 			} else if shard.ClusterID != wakuM.clusterID {
 				err = errors.New("different clusterID reported")
 			}
 		} else {
 			// Only disconnect from peers if they support the protocol
-			// TODO: open a PR in go-libp2p to create a var with this error to not have to compare strings but use errors.Is instead
-			if strings.Contains(err.Error(), "protocols not supported") {
+			if isProtocolNotSupported(err) {
```

Comment:
Same comment as above, if clusterID is !=0 for localNode, then shouldn't we disconnect peers that don't support the metadata protocol?

Trying to understand what would be the case where we would need to be connected to such peer.


Also, this check seems duplicate to the one already done in lines 103-105 in this PR.

Reply:
fixed in last commit